tests/sources/test_ads.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+
+# (C) Copyright 2020 ECMWF.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+#
+
+import pytest
+
+from earthkit.data import from_source
+from earthkit.data.testing import NO_CDS
```

Comment:
`NO_CDS` is defined as:

```python
NO_CDS = not os.path.exists(os.path.expanduser("~/.cdsapirc"))
```

When we want to access both the CDS and the ADS via `cdsapi` we need to create a separate rc file for each. For example I use `~/.cdaspirc` for CDS and `~/.adsapirc` for ads. So using `NO_CDS` as defined here is not adequate to detect whether we have access to ADS. It can also happen that "~/.cdaspirc" contains the key for the ADS access! 

The bottom line is that, you need to define both `NO_CDS` and  `NO_ADS` with a more elaborate check on the availability of the given service.